tests/pyfunc/test_docker.py

```diff
@@ -0,0 +1,128 @@
+import difflib
+import os
+import shutil
+from dataclasses import dataclass
+from pathlib import Path
+from typing import Optional
+from unittest import mock
+
+import pytest
+import sklearn
+import sklearn.neighbors
+
+import mlflow
+from mlflow.models.flavor_backend_registry import get_flavor_backend
+from mlflow.utils.env_manager import CONDA, VIRTUALENV
+from mlflow.version import VERSION
+
+_MLFLOW_ROOT = os.path.abspath(os.path.join(os.path.dirname(__file__), "..", ".."))
+_RESOURCE_DIR = os.path.join(_MLFLOW_ROOT, "tests", "resources", "dockerfile")
+
+
+def compare_dockerfiles(actual_dockerfile_path: Path, expected_dockerfile_path: Path):
```

Comment:
Not a blocker, but a function name like `assert_xxx_equal` would be more helpful for code readers because that would tell us what this function **_validates_**.